feat: Add option to specify media types

afn1988 opened the pull request:
https://swagger.io/docs/specification/v3_0/media-types/

afn1988 committed:
feat: Add option to specify media types

https://swagger.io/docs/specification/v3_0/media-types/

rjbma commented on src/index.ts:
```diff
@@ -7,8 +7,15 @@ import crypto from "crypto";
-type JsonSchema = Record<string, unknown>;
+type SchemaObject = Record<string, unknown>;
 type SecurityRequirement = { name: string; scopes: string[] };
+type MediaType =
```
There are a lot more media types, so this should probably just be `string`

afn1988 replied:
We've decided to add text/plain
type MediaType =
  | "application/json"
  | "application/xml"
  | "application/x-ndjson"
  | "text/plain";

rjbma commented on src/index.ts:
```diff
@@ -482,26 +489,46 @@ const oas = {
+        };
+      } else {
+        // Different schemas for different media types
+        for (const mediaType of Object.keys(route.validate.body)) {
```
replace this with a `Object.keys(route.validate.body).forEach`, mainly for consistency

rjbma commented on src/index.ts:
```diff
@@ -482,26 +489,46 @@ const oas = {
-            schema: route.validate.body.toJSonSchema(options),
-            examples: route.validate.body._examples,
-          },
+        requestBody: route.validate?.body && {
```
nested `requestBody` inside another `requestBody`

rjbma commented on src/index.ts:
```diff
@@ -535,15 +562,32 @@ const routeToJsonSchema = (route: Route, options: Options): JsonSchema => {
+        const schema =
+          response.schema[mediaType as keyof typeof response.schema];
+        if (schema) {
+          // Add this null check
```
what's the purpose of this comment?

afn1988 replied:
it was a TODO...

rjbma commented on src/index.ts:
```diff
@@ -535,15 +562,32 @@ const routeToJsonSchema = (route: Route, options: Options): JsonSchema => {
-      },
+      };
+    } else {
+      for (const mediaType of Object.keys(response.schema)) {
```
replace this with a `Object.keys(route.validate.body).forEach`, mainly for consistency

rjbma commented on src/index.ts:
```diff
@@ -727,7 +773,7 @@ function collectRefSchemasForSchema(schema: Schema | undefined): FixedSchema[] {
 function collectRefSchemasForRoute(route: Route): FixedSchema[] {
   if ("validate" in route) {
-    const schemas: (Schema | undefined)[] = [
+    const schemas: (Schema | MediaTypeSchemaMap | undefined)[] = [
```
I think we can remove the type signature, typescript should be able to correctly infer the type. Try to leave as

    const schemas = [

afn1988 committed:
feat: add text/plain media type and fix MediaTypeSchemaMap narrowing

- Add  to the  union
- Fix TypeScript narrowing in  and  by casting to  before indexing, and switch  loops to  for consistency
- Fix duplicate  nesting
- Remove stale  comment
- Trailing comma fixes throughout
